### PR TITLE
Validate migrateConnection argument when a custom connection is setted

### PR DESCRIPTION
**Pull request by waldemarnt**
That will fix this issue https://github.com/CakeDC/migrations/issues/212 it's a bug I guess, not a feature how he told in the issue.
Tnx guys

### CODE REVIEW

File: Console/Command/MigrationShell.php [hunk 1]
```diff
@@ -118,6 +121,32 @@ public function startup() {
...
+ */
+	protected function _connectionNamesEnum(){
+		return array_keys($this->ConnectionManager->enumConnectionObjects());
+	}
+
+/**
+ * Check if have a migration connect parameter when connection is custom
+ * @return String
+ */
+	protected function _startMigrationConnection(){
```
Comment (skie) [inline Console/Command/MigrationShell.php]: code formatting issue

File: Console/Command/MigrationShell.php [hunk 2]
```diff
@@ -23,6 +23,10 @@
 class MigrationShell extends AppShell {
 
 /**
+ * Call the ConnectionManager Model
+ */
+	public $uses = array('ConnectionManager');
```
Comment (steinkel) [inline Console/Command/MigrationShell.php]: hmm?

ConnectionManager is not a Model

File: Console/Command/MigrationShell.php [hunk 3]
```diff
@@ -118,6 +121,32 @@ public function startup() {
 	}
 
 /**
+ * Get connection names list
+ * @return Array
+ */
+	protected function _connectionNamesEnum() {
+		return array_keys($this->ConnectionManager->enumConnectionObjects());
```
Comment (steinkel) [inline Console/Command/MigrationShell.php]: enumConnectionObjects is a static function https://github.com/cakephp/cakephp/blob/2.6/lib/Cake/Model/ConnectionManager.php#L192

File: Console/Command/MigrationShell.php [hunk 4]
```diff
@@ -118,6 +121,32 @@ public function startup() {
 	}
 
 /**
+ * Get connection names list
+ * @return Array
```
Comment (deizel) [inline Console/Command/MigrationShell.php]: Blank line before, lowercase type and (optional) description. For example:

```patch
- * @return Array
+ * 
+ * @return array The array of connection names.

File: Console/Command/MigrationShell.php [hunk 5]
```diff
@@ -118,6 +121,32 @@ public function startup() {
...
+ */
+	protected function _connectionNamesEnum() {
+		return array_keys($this->ConnectionManager->enumConnectionObjects());
+	}
+
+/**
+ * Check if have a migration connect parameter when connection is custom
+ * @return String
```
Comment (deizel) [inline Console/Command/MigrationShell.php]: Same as above.

File: Console/Command/MigrationShell.php (hunk 3 above)
Comment (waldemarnt) [inline Console/Command/MigrationShell.php]: What is the problem of the way that I used?
tnx

Comment (deizel) [inline Console/Command/MigrationShell.php]: See here: https://github.com/cakephp/cakephp/blob/2.6/lib/Cake/Console/Command/Task/DbConfigTask.php#L356

File: Console/Command/MigrationShell.php [hunk 6]
```diff
@@ -118,6 +124,34 @@ public function startup() {
...
+ * @return array The list of connection names
+ */
+	protected function _connectionNamesEnum() {
+		return array_keys(ConnectionManager::enumConnectionObjects());
+	}
+
+/**
+ * Set a migrationConnection
+ *
+ * @return String
```
Comment (deizel) [inline Console/Command/MigrationShell.php]: Lowercase `string` and (optional) description. Example:

```php
* @return string The name of the migration connection.
```

File: Console/Command/MigrationShell.php [hunk 7]
```diff
@@ -23,6 +23,13 @@
 class MigrationShell extends AppShell {
 
 /**
+ * ConnectionManager instance
+ *
+ * @var ConnectionManager
+ */
+	public $uses = array('ConnectionManager');
```
Comment (deizel) [inline Console/Command/MigrationShell.php]: This can be removed now that you are calling it statically.

File: Console/Command/MigrationShell.php [hunk 8]
```diff
@@ -15,6 +15,7 @@
 App::uses('MigrationVersion', 'Migrations.Lib');
 App::uses('String', 'Utility');
 App::uses('ClassRegistry', 'Utility');
+App::uses('ConnectionManager','Model');
```
Comment (deizel) [inline Console/Command/MigrationShell.php]: Space after comma.

File: Console/Command/MigrationShell.php [hunk 9]
```diff
@@ -79,9 +80,8 @@ public function startup() {
 			$this->connection = $this->params['connection'];
 		}
 
-		if (!empty($this->params['migrationConnection'])) {
-			$this->migrationConnection = $this->params['migrationConnection'];
-		}
+		$this->migrationConnection = $this->_startMigrationConnection();
+
```
Comment (deizel) [inline Console/Command/MigrationShell.php]: Let's remove this new line also.

**Comment by deizel** [comment]
Looking good, just a few last things and I think this will be good to merge. :+1:

**Pull request merged**